pkg/server/registry/apigroups/acorn/jobs/translator.go

```diff
@@ -0,0 +1,130 @@
+package jobs
+
+import (
+	"context"
+	"strings"
+
+	"github.com/acorn-io/baaah/pkg/router"
+	mtypes "github.com/acorn-io/mink/pkg/types"
+	apiv1 "github.com/acorn-io/runtime/pkg/apis/api.acorn.io/v1"
+	"github.com/acorn-io/runtime/pkg/labels"
+	"github.com/acorn-io/runtime/pkg/namespace"
+	"github.com/acorn-io/runtime/pkg/publicname"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	klabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apiserver/pkg/storage"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Translator struct {
+	client kclient.Client
+}
+
+func (t *Translator) FromPublicName(ctx context.Context, namespace, name string) (string, string, error) {
+	appName, jobName := publicname.Split(name)
+
+	app := &apiv1.App{}
+	err := t.client.Get(ctx, router.Key(namespace, appName), app)
+	if err != nil {
+		return namespace, name, err
+	}
+
+	namespace = app.Status.Namespace
+	return namespace, strings.Split(jobName, ":")[0], nil
```

Comment:
I'm a little confused...how can the jobName have a `:` in it?

Reply:
This line was a causality of the switch to a custom strategy. However, I believe this is only `container` specific and not relevant to `jobs`. As a result this would've likely been removed anyway.